UsefulProteomicsDatabases/PtmListLoaderGeneral.cs

```diff
@@ -0,0 +1,386 @@
+                                {
+                                    yield return new ModificationGeneral(_Id, _Accession, _ModificationType, _FeatureType, motif, _Position, _ChemicalFormula, _MonoisotopicMass, _DatabaseReference, _TaxonomicRange, _Keywords, _NeutralLosses, _DiagnosticIons, _FileOrigin);
+                                }
+                            else
+                            {
+                                yield return new ModificationGeneral(_Id, _Accession, _ModificationType, _FeatureType, null, _Position, _ChemicalFormula, _MonoisotopicMass, _DatabaseReference, _TaxonomicRange, _Keywords, _NeutralLosses, _DiagnosticIons, _FileOrigin);
+                            }
+                            break;
+
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+
+
+        public static DissociationType? ModDissociationType(string modType)
+        {
+            switch (modType)
+            {
+                case "Any":
+                    return DissociationType.AnyActivationType;
+                case "CID":
+                    return DissociationType.CID;
+                case "MPD":
+                    return DissociationType.MPD;
+                case "ECD":
+                    return DissociationType.ECD;
+                case "PQD":
+                    return DissociationType.PQD;
+                case "ETD":
+                    return DissociationType.ETD;
+                case "HCD":
+                    return DissociationType.HCD;
+                case "SA":
+                    return DissociationType.SA;
+                case "PTR":
+                    return DissociationType.PTR;
+                case "NETD":
+                    return DissociationType.NETD;
+                case "NPTR":
+                    return DissociationType.NPTR;
+                case "ISCID":
+                    return DissociationType.ISCID;
+                default:
+                    return null;            
+             }
+        }
+
+        public static Dictionary<DissociationType, List<double>> DiagnosticIonsAndNeutralLosses(string oneEntry)
+        {
+            Dictionary<DissociationType, List<double>> dAndNDictionary = new Dictionary<DissociationType, List<double>>();
+
+            try
+            {
+                string[] nlOrDiEntries = oneEntry.Split(new string[] { " or " }, StringSplitOptions.None);
+                foreach (string nlOrDiEntry in nlOrDiEntries)
+                {
+                    string[] entryKeyValue = nlOrDiEntry.Split(new string[] { ":" }, StringSplitOptions.RemoveEmptyEntries);
+                    if (entryKeyValue.Length == 1) // assume there is no dissociation type listed and only formula or mass are supplied
+                    {
+                        try // see if dictionary already contains key AnyActivationType
```

Comment:
Can you change this to use the boolean return value of `.TryGetValue()`? That would eliminate these try-catch blocks, which can be problematic if Exceptions other than the ones you anticipated happen.